### PR TITLE
feat(notebooks): add support for git workspaces PS-13866

### PR DESCRIPTION
**Pull request by bbatha**
I also have chosen to explicitly not support workspace uploads yet. Those will come in another ticket/pr when we can add API support for them.

### CODE REVIEW

File: tests/functional/test_notebooks.py [hunk 1]
```diff
@@ -59,8 +59,9 @@ class TestNotebooksCreate(object):
         "machineType": "P5000",
         "container": "jupyter/notebook",
         "clusterId": "321",
-        'isPreemptible': False,
-        'isPublic': False,
+        "isPreemptible": False,
+        "isPublic": False,
+        "workspace": "none",
```
Comment (BartoszCki) [inline tests/functional/test_notebooks.py:64]: I wanted to finally drop passing "none" to workspace. The PR to that is waiting for review. Replace this with some URL or remove the line, please.

File: tests/functional/test_notebooks.py [hunk 2]
```diff
@@ -108,6 +109,7 @@ class TestNotebooksCreate(object):
         "shutdownTimeout": 8,
         "isPreemptible": True,
         "isPublic": False,
+        "workspace": "none",
```
Comment (BartoszCki) [inline tests/functional/test_notebooks.py:112]: The same as above

**Comment by bbatha** [comment]
@BartoszCki requested changes made

File: tests/functional/test_notebooks.py (hunk 1 above)
Comment (bbatha) [reply tests/functional/test_notebooks.py:64]: After attempting the change, I realized that we still want this test for `"none"`. This is the expected request json the notebook endpoints depend on sending an explicit 'none' for no workspace and that can't really change until we retire v1. I have updated the all options test to use an explicit url.

**Pull request merged**